_source/_posts/2022-07-11-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,731 @@
+
+This is the Spring Cloud Config file that the JHipster registry will serve to the entire microservice network.
+
+`kubernetes/registry-k8s/application-configmap.yml`
+
+```yaml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+# app specific configuration
+jhipster-registry.yml: |-
+  ...
+```
+
+You also need to configure the registry to use OAuth. In the `kubernetes/registry-k8s/jhipster-registry.yml` file, add `oauth2` to the values under `SPRING_PROFILES_ACTIVE` (as shown below).
+
+`kubernetes/registry-k8s/jhipster-registry.yml`
+
+```yaml
+spec:
+  containers:
+    - name: jhipster-registry
+      env:
+        - name: SPRING_PROFILES_ACTIVE
+          value: prod,k8s,oauth2
+```
+
+## Create an Amazon Elastic Kubernetes (EKS) cluster
+
+Now you are going to create the EKS cluster on AWS using the [eksctl](https://eksctl.io/) CLI.
+
+You should have the AWS CLI and `eksctl` installed and configured already. Use the following command to create the Kubernetes cluster for the microservice. After you run the command, go get some coffee or lunch. This command is *slooooow*.
+
+```bash
+eksctl create cluster --name okta-k8s \
+    --region us-west-2 \
+    --nodegroup-name okta-k8s-nodes \
+    --node-type t2.xlarge \
+    --nodes 2
+```
+
+**This will take several minutes to run.**
+
+**This cluster costs money as long as it is running.** If you only keep it running for the hour or so you need to run the tutorial, the costs are small. **If you forget and leave it running for a month or two, it could cost a lot more. Don't forget to delete it when you are done!**
+
+You are using a managed cluster with EC2 nodes. You are not using the newer, Fargate managed clusters because when I tried Fargate I ran into problems. There are reports of a compatibility problem between Fargate and Eureka. See [this google groups thread](https://groups.google.com/g/jhipster-dev/c/148uESs6vns) and [this GitHub repository](https://github.com/jussiseppala/eurekafargatesample).
+
+You can look at [the AWS docs on managed node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) for more in-depth information. Basically, though, being managed means that you do not need to separately provision and register the EC2 instances in the node group. The EC2 instances and Kubernetes nodes are packaged together and managed as a unit. Unmanaged node groups allow you to separate the node group management from the EC2 instances, allowing for finer control but also requiring more direct involvement in instance management.
+
+## Using AWS CloudFormation to troubleshoot problems
```

Comment:
```suggestion
## Troubleshoot problems using AWS CloudFormation
```

cc: Havi: nit, but this suggestion shortens the number of chars in this header to prevent the dynamic line wrap of the link icon when hovering over the header.

(There is probably something we can do with the CSS longer term, but take it or leave it 😄)